File: backend/rwm/src/main/java/com/rwm/dto/request/UpdateProfileRequest.java
package com.rwm.dto.request;

import lombok.Data;

/**
 * 更新用户资料请求
 */
@Data
public class UpdateProfileRequest {
    
    private String email;
    
    private String firstName;
    
    private String lastName;
    
    private String gender; // MALE, FEMALE, OTHER
    
    private Integer age;
    
    private String phoneNumber;
    
    private String fitnessLevel; // BEGINNER, INTERMEDIATE, ADVANCED
    
    private Double height;
    
    private Double weight;
    
    private String weeklyAvailability;
}

File: app/src/main/java/com/example/myapplication/landr/profile/UpdateProfileRequest.kt
package com.example.myapplication.landr.profile

data class UpdateProfileRequest(
    val email: String? = null,
    val firstName: String? = null,
    val lastName: String? = null,
    val gender: String? = null,
    val age: Int? = null,
    val phoneNumber: String? = null,
    val fitnessLevel: String? = null,
    val height: Double? = null,
    val weight: Double? = null,
    val weeklyAvailability: String? = null
)

